File: examples/projects/benchmarks/btree/btree.go
// 基准: btree —— 深度 15 完全二叉树（数组存储）+ 递归遍历求和
// 与 btree.rl 逻辑严格一致。输出 = 2166712927200
package main

import "fmt"

const total = 65535

func treeSum(a []int64, idx int64) int64 {
	if idx >= total {
		return 0
	}
	left := treeSum(a, idx*2+1)
	right := treeSum(a, idx*2+2)
	return a[idx] + left + right
}

func main() {
	a := make([]int64, total)
	for i := int64(0); i < total; i++ {
		a[i] = i*1009 + 17
	}
	fmt.Println(treeSum(a, 0))
}

File: examples/projects/benchmarks/btree/btree.swift
// 基准: btree —— 深度 15 完全二叉树（数组存储）+ 递归遍历求和
// 与 btree.rl 逻辑严格一致。输出 = 2166712927200
let total: Int64 = 65535

func treeSum(_ a: [Int64], _ idx: Int64, _ total: Int64) -> Int64 {
    if idx >= total { return 0 }
    let left = treeSum(a, idx * 2 + 1, total)
    let right = treeSum(a, idx * 2 + 2, total)
    return a[Int(idx)] + left + right
}

var a = [Int64]()
a.reserveCapacity(Int(total))
for i in 0..<total {
    a.append(i * 1009 + 17)
}
print(treeSum(a, 0, total))
